x/blob/types/payforblob.go

```diff
@@ -195,14 +240,21 @@ func CreateCommitments(blobs []*Blob) ([][]byte, error) {
 	return commitments, nil
 }
 
-// ValidatePFBComponents performs basic checks over the components of one or more PFBs.
+// ValidateBlobs performs basic checks over the components of one or more PFBs.
 func ValidateBlobs(blobs ...*Blob) error {
 	if len(blobs) == 0 {
 		return ErrNoBlobs
 	}
 
 	for _, blob := range blobs {
-		err := ValidateBlobNamespaceID(blob.NamespaceId)
+		if blob.NamespaceVersion > math.MaxUint8 {
+			return fmt.Errorf("namespace version %d is too large", blob.NamespaceVersion)
+		}
+		ns, err := appns.New(uint8(blob.NamespaceVersion), blob.NamespaceId)
+		if err != nil {
+			return err
+		}
+		err = ns.ValidateBlobNamespace()
```

Comment:
[optional comment/question]
could we combine the validation the the New method? If so, do we gain anything from that?

Reply:
we could if we created a new function:
```go
NewBlobNamespace
```

I don't think it makes sense to enforce this validation on all usage of `New` because all reserved namespaces will fail the `ValidateBlobNamespace` check

Reply:
I see thanks for the clarification

Reply:
out of curiosity, where are we using the new function to create reserved namespaces?

Reply:
1. https://github.com/rootulp/celestia-app/blob/41c541386b43d993bbd32e92f37cf60d2c94f20e/app/process_proposal.go#L48
2. https://github.com/rootulp/celestia-app/blob/41c541386b43d993bbd32e92f37cf60d2c94f20e/pkg/shares/split_sparse_shares.go#L32
3. https://github.com/rootulp/celestia-app/blob/41c541386b43d993bbd32e92f37cf60d2c94f20e/x/blob/client/cli/payforblob.go#L39
+ usage in 3 more files